velox/dwio/parquet/tests/ParquetReaderTestBase.h

```diff
@@ -51,6 +51,12 @@ class ParquetReaderTestBase : public testing::Test {
     return ROW({"int", "bigint"}, {INTEGER(), BIGINT()});
   }
 
+  static RowTypePtr unsignedIntSchema() {
+    return ROW(
+        {"uint8", "uint16", "uint32", "uint64"},
+        {SMALLINT(), INTEGER(), BIGINT(), DECIMAL(20, 0)});
```

Comment:
You only used a fixed schema. Can you also try more possible combinations? E.g. 
`{SMALLINT(), INTEGER(), INTEGER(), DECIMAL(20, 0)}`
and then the expected vector for the 3rd column would be
`vectorMaker_->flatVector<uint32_t>({4294967295, 2000000000, 3000000000}),`
Please make sure all T - TVector size combinations are tested. Thanks!

Reply:
hey @yingsu00 thank you very much for the suggestion. Below is my concern, hope we can discuss.

firstly, the case you examples here 

```
{"uint8", "uint16", "uint32", "uint64"},
{SMALLINT(), INTEGER(), INTEGER(), DECIMAL(20, 0)}
```

actually i thought it may not be a valid case, it means the reader requesting same data type (INTEGER) when reading different data type column unit16 and uint32 from parquet file, here it causes the valueSize_ for both uint16 and uint32 be set to 4 (parquet store uint16 and uint64 with 32 bitswidth) and the request datatype for both is INTEGER, it looks not right.

even we can leverage parquet logic type's bitwidth property to distinguish them, it still looks doesn't make sense
why client would ask same INTEGER data type for different unsigned int type clomun.

secondly, when client reads a parquet file, the appropriate assumption is that it should requests different data type for columns with different unsigned int data type, that aligns with how Spark handles the situation: when spark reads usigned integers from parquet, it consistently requests uint8 as int16, uint16 as int32, unit32 as int64 and uint64 as int128.
So the proposed changes should work fine when integrating with spark.

thirdly, using SMALLINT() for accessing uint8, INTEGER() for accessing uint16,  BIGINT for accessing uint32, DCIMAL(20, 0) for accessing uint64 in my opinion is a correct schemas choice. Other combinations may not align with the typical use case requirements.

consider using INTEGER or HUGINT when accessing uint32, the related native c type range ethier too small or too big.

However, if you believe it is mandatory requirments that client may still use other combinations as
```{TYNYINT(), SMALLINT(), INTEGER(), BIGINT() }```
```{INTEGER(), BIGINT(), HUGINT(), DECIMAL(20, 0)}```
to accessing data, and it is valid.

then I would happy to make changes in this PR to try to support those combinations.

or if you agree that other part combinations is not a strongly required, then we can initially proceed with the current changes to unblock the integration with Spark's Parquet unsigned integers, and address additional support in subsequent PRs.

Reply:
> {"uint8", "uint16", "uint32", "uint64"},
{SMALLINT(), INTEGER(), INTEGER(), DECIMAL(20, 0)}
> actually i thought it may not be a valid case,

It should be valid case. https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#unsigned-integers clearly stated that INT(32, false) must annotate an int32 primitive type. A 32 bit integer in Parquet file can identify a uint32 value. I tried the test and it worked.

Reply:
done to add more valid combinations